jerry-core/ecma/builtin-objects/ecma-builtin-atomics.c

```diff
@@ -0,0 +1,385 @@
+
+/**
+ * The Atomics object's 'wait' routine
+ *
+ * See also:
+ *          24.4.11
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_atomics_wait (ecma_value_t arg1, /**< typedArray argument */
+                           ecma_value_t arg2, /**< index argument */
+                           ecma_value_t arg3, /**< value argument */
+                           ecma_value_t arg4) /**< timeout argument */
+{
+  JERRY_UNUSED (arg1);
+  JERRY_UNUSED (arg2);
+  JERRY_UNUSED (arg3);
+  JERRY_UNUSED (arg4);
+
+  return ecma_make_uint32_value (0);
+} /* ecma_builtin_atomics_wait */
+
+/**
+ * The Atomics object's 'notify' routine
+ *
+ * See also:
+ *          24.4.12
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_atomics_notify (ecma_value_t arg1, /**< typedArray argument */
+                             ecma_value_t arg2, /**< index argument */
+                             ecma_value_t arg3) /**< count argument */
+{
+  JERRY_UNUSED (arg1);
+  JERRY_UNUSED (arg2);
+  JERRY_UNUSED (arg3);
+
+  return ecma_make_uint32_value (0);
+} /* ecma_builtin_atomics_notify */
+
+/**
+ * The Atomics object's 'or' routine
+ *
+ * See also:
+ *          24.4.13
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_atomics_xor (ecma_value_t arg1, /**< typedArray argument */
+                          ecma_value_t arg2, /**< index argument */
+                          ecma_value_t arg3) /**< value argument */
+{
+  return ecma_atomic_read_modify_write (arg1, arg2, arg3, ECMA_ATOMICS_XOR);
+} /* ecma_builtin_atomics_xor */
+
+/**
+ * Dispatcher of the built-in's routines
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_atomics_dispatch_routine (uint8_t builtin_routine_id, /**< built-in wide routine identifier */
+                                       ecma_value_t this_arg, /**< 'this' argument value */
+                                       const ecma_value_t arguments_list_p[], /**< list of arguments
+                                                                             *   passed to routine */
+                                        uint32_t arguments_number) /**< length of arguments' list */
+{
+  JERRY_UNUSED (this_arg);
+  ecma_value_t arg1 = (arguments_number > 0) ? arguments_list_p[0] : ECMA_VALUE_UNDEFINED;
+  ecma_value_t arg2 = (arguments_number > 1) ? arguments_list_p[1] : ECMA_VALUE_UNDEFINED;
+  ecma_value_t arg3 = (arguments_number > 2) ? arguments_list_p[2] : ECMA_VALUE_UNDEFINED;
+  ecma_value_t arg4 = (arguments_number > 3) ? arguments_list_p[3] : ECMA_VALUE_UNDEFINED;
+
+  switch (builtin_routine_id)
+  {
+    case ECMA_ATOMICS_ROUTINE_ADD:
+    {
+      return ecma_builtin_atomics_add (arg1, arg2, arg3);
```

Comment:
I would prefer to not create these one-liner functions, just call the appropriate form of ecma_atomic_read_modify_write.  I am curious about others opinios as well. A very simplified example:

```
type_t type = default;
switch (builtin_routine_id)
{
  case add: type = add; break;
  case compare_exchange: ecma_compare_exchange(...); return;
}
ecma_atomic_read_modify_write(..., type)
```

So the read_modify variants are collected at the end. Otherwise the switch returns early.